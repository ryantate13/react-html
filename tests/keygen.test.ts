import {keygen} from '../src';

describe('keygen', () => {
    it('renders with children', () => {
        expect(keygen({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(keygen({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(keygen({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(keygen()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(keygen({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(keygen({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(keygen({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(keygen({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(keygen({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(keygen({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(keygen({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with autoFocus=true', () => {
        expect(keygen({autoFocus: true})).toMatchSnapshot();
    });

    it('renders with autoFocus=false', () => {
        expect(keygen({autoFocus: false})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(keygen({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(keygen({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(keygen({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(keygen({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(keygen({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(keygen({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(keygen({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(keygen({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(keygen({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(keygen({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(keygen({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(keygen({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(keygen({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(keygen({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(keygen({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(keygen({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(keygen({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(keygen({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(keygen({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(keygen({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(keygen({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(keygen({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(keygen({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(keygen({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(keygen({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(keygen({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(keygen({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(keygen({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(keygen({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(keygen({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(keygen({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(keygen({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(keygen({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(keygen({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(keygen({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(keygen({translate: "no"})).toMatchSnapshot();
    });
});