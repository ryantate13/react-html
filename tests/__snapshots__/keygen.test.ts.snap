// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keygen renders with accessKey="a,b,c,d" 1`] = `
<keygen
  accessKey="a,b,c,d"
/>
`;

exports[`keygen renders with autoCapitalize="characters" 1`] = `
<keygen
  autoCapitalize="characters"
/>
`;

exports[`keygen renders with autoCapitalize="none" 1`] = `
<keygen
  autoCapitalize="none"
/>
`;

exports[`keygen renders with autoCapitalize="off" 1`] = `
<keygen
  autoCapitalize="off"
/>
`;

exports[`keygen renders with autoCapitalize="on" 1`] = `
<keygen
  autoCapitalize="on"
/>
`;

exports[`keygen renders with autoCapitalize="sentences" 1`] = `
<keygen
  autoCapitalize="sentences"
/>
`;

exports[`keygen renders with autoCapitalize="words" 1`] = `
<keygen
  autoCapitalize="words"
/>
`;

exports[`keygen renders with children 1`] = `
<keygen>
  TEST
</keygen>
`;

exports[`keygen renders with className="foo bar baz" 1`] = `
<keygen
  className="foo bar baz"
/>
`;

exports[`keygen renders with contentEditable="false" 1`] = `
<keygen
  contentEditable="false"
/>
`;

exports[`keygen renders with contentEditable="true" 1`] = `
<keygen
  contentEditable="true"
/>
`;

exports[`keygen renders with dir="auto" 1`] = `
<keygen
  dir="auto"
/>
`;

exports[`keygen renders with dir="ltr" 1`] = `
<keygen
  dir="ltr"
/>
`;

exports[`keygen renders with dir="rtl" 1`] = `
<keygen
  dir="rtl"
/>
`;

exports[`keygen renders with draggable="false" 1`] = `
<keygen
  draggable="false"
/>
`;

exports[`keygen renders with draggable="true" 1`] = `
<keygen
  draggable="true"
/>
`;

exports[`keygen renders with empty attributes 1`] = `<keygen />`;

exports[`keygen renders with hidden="false" 1`] = `
<keygen
  hidden={false}
/>
`;

exports[`keygen renders with hidden="true" 1`] = `
<keygen
  hidden={true}
/>
`;

exports[`keygen renders with id="TEST" 1`] = `
<keygen
  id="TEST"
/>
`;

exports[`keygen renders with inputMode="decimal" 1`] = `
<keygen
  inputMode="decimal"
/>
`;

exports[`keygen renders with inputMode="email" 1`] = `
<keygen
  inputMode="email"
/>
`;

exports[`keygen renders with inputMode="none" 1`] = `
<keygen
  inputMode="none"
/>
`;

exports[`keygen renders with inputMode="numeric" 1`] = `
<keygen
  inputMode="numeric"
/>
`;

exports[`keygen renders with inputMode="search" 1`] = `
<keygen
  inputMode="search"
/>
`;

exports[`keygen renders with inputMode="tel" 1`] = `
<keygen
  inputMode="tel"
/>
`;

exports[`keygen renders with inputMode="text" 1`] = `
<keygen
  inputMode="text"
/>
`;

exports[`keygen renders with inputMode="url" 1`] = `
<keygen
  inputMode="url"
/>
`;

exports[`keygen renders with is="foo" 1`] = `
<keygen
  is="foo"
/>
`;

exports[`keygen renders with itemID="https://google.com/" 1`] = `
<keygen
  itemID="https://google.com/"
/>
`;

exports[`keygen renders with itemProp="https://google.com/" 1`] = `
<keygen
  itemProp="https://google.com/"
/>
`;

exports[`keygen renders with itemRef="foo bar" 1`] = `
<keygen
  itemRef="foo bar"
/>
`;

exports[`keygen renders with itemScope="false" 1`] = `
<keygen
  itemScope={false}
/>
`;

exports[`keygen renders with itemScope="true" 1`] = `
<keygen
  itemScope={true}
/>
`;

exports[`keygen renders with itemType="https://google.com/" 1`] = `
<keygen
  itemType="https://google.com/"
/>
`;

exports[`keygen renders with lang="" 1`] = `
<keygen
  lang=""
/>
`;

exports[`keygen renders with lang="es-419" 1`] = `
<keygen
  lang="es-419"
/>
`;

exports[`keygen renders with slot="TEST" 1`] = `
<keygen
  slot="TEST"
/>
`;

exports[`keygen renders with spellCheck="false" 1`] = `
<keygen
  spellCheck="false"
/>
`;

exports[`keygen renders with spellCheck="true" 1`] = `
<keygen
  spellCheck="true"
/>
`;

exports[`keygen renders with style="[object Object]" 1`] = `
<keygen
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`keygen renders with tabIndex="1" 1`] = `
<keygen
  tabIndex={1}
/>
`;

exports[`keygen renders with title="TEST" 1`] = `
<keygen
  title="TEST"
/>
`;

exports[`keygen renders with translate="no" 1`] = `
<keygen
  translate="no"
/>
`;

exports[`keygen renders with translate="yes" 1`] = `
<keygen
  translate="yes"
/>
`;

exports[`keygen renders without attributes 1`] = `<keygen />`;
